internal/core/src/index/IndexFactory.cpp

```diff
@@ -78,6 +79,202 @@ IndexFactory::CreatePrimitiveScalarIndex<std::string>(
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::bin1>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        case milvus::DataType::VECTOR_FLOAT:
+            resource = knowhere::IndexStaticFaced<
+                knowhere::fp32>::EstimateLoadResource(index_type,
+                                                      index_version,
+                                                      index_size_gb,
+                                                      config);
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::fp32>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        case milvus::DataType::VECTOR_FLOAT16:
+            resource = knowhere::IndexStaticFaced<
+                knowhere::fp16>::EstimateLoadResource(index_type,
+                                                      index_version,
+                                                      index_size_gb,
+                                                      config);
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::fp16>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        case milvus::DataType::VECTOR_BFLOAT16:
+            resource = knowhere::IndexStaticFaced<
+                knowhere::bf16>::EstimateLoadResource(index_type,
+                                                      index_version,
+                                                      index_size_gb,
+                                                      config);
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::bf16>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        case milvus::DataType::VECTOR_SPARSE_FLOAT:
+            resource = knowhere::IndexStaticFaced<
+                knowhere::fp32>::EstimateLoadResource(index_type,
+                                                      index_version,
+                                                      index_size_gb,
+                                                      config);
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::fp32>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        default:
+            PanicInfo(
+                milvus::DataTypeInvalid,
+                fmt::format(
+                    "invalid data type to estimate index load resource: {}",
+                    field_type));
+    }
+
+    LoadResourceRequest request{};
+
+    request.has_raw_data = has_raw_data;
+    request.final_disk_cost = resource.value().diskCost;
+    request.final_memory_cost = resource.value().memoryCost;
+    if (knowhere::UseDiskLoad(index_type, index_version) || mmaped) {
+        request.max_disk_cost = resource.value().diskCost;
+        request.max_memory_cost =
```

Comment:
should it be esource.value().memoryCost +  download_buffer_size_gb

Reply:
> should it be esource.value().memoryCost + download_buffer_size_gb

The download and deserialization of the vector index are two separate stages. During deserialization, the memory buffer used for downloading has already been released, so the maximum memory usage is the larger of the two stages.